vscode-trace-extension/src/utils/trace-extension-logger.ts

```diff
@@ -0,0 +1,182 @@
+        }
+    }
+
+    /**
+     * Disposes the log channel and all of its resources
+     */
+    disposeChannel(): void {
+        this.logChannel.dispose();
+        this.channelDisposed = true;
+    }
+
+    /**
+     * Adds a log entry to the log channel with an optional tag
+     *
+     * @param message message to be appended to the log channel
+     * @param tag appended to the front of the message (i.e. [$tag] $message)
+     */
+    addLogMessage(message: string, tag?: string): void {
+        let fullMessage: string;
+        if (tag) {
+            fullMessage = '[' + tag + '] ' + message;
+        } else {
+            fullMessage = message;
+        }
+        this.addLogPanelMessage(fullMessage);
+    }
+
+    /**
+     * Adds a log entry for an error.
+     * Prints the error message and the stack
+     *
+     * @param e error to be appended to the log channel
+     */
+    logError(e: Error): void {
+        this.addLogMessage(e.message);
+        if (e.stack) {
+            this.addLogMessage(e.stack);
+        }
+    }
+
+    /**
+     * Shows a information message to user as a notification and
+     * adds the message to log the log channel
+     *
+     * @param message information message to show to the user and add to the log channel
+     * @param options Configures the behavior of the message
+     * @returns void promise
+     */
+    async showInfo(message: string, options?: vscode.MessageOptions): Promise<void> {
+        const fullMessage = 'Info: ' + message;
+        vscode.window.showInformationMessage(fullMessage, options || { modal: false });
+        return this.addLogMessage(fullMessage);
+    }
+
+    /**
+     * Shows a warning message to user as a notification and
+     * adds the message to log the log channel
+     *
+     * @param message warning message to show to the user and add to the log channel
+     * @param options Configures the behavior of the message
+     * @returns void promise
+     */
+    async showWarning(message: string, options?: vscode.MessageOptions): Promise<void> {
+        const fullMessage = 'Warning: ' + message;
+        vscode.window.showWarningMessage(fullMessage, options || { modal: false });
+        return this.addLogMessage(fullMessage);
+    }
+
+    /**
+     * Shows an error message to user as a notification and
+     * adds the message to log the log channel
+     *
+     * @param message error message to show to the user and add to the log channel
+     * @param options Configures the behavior of the message
+     * @returns void promise
+     */
+    async showError(message: string, options?: vscode.MessageOptions): Promise<void> {
+        const fullMessage = 'Error: ' + message;
+        vscode.window.showErrorMessage(fullMessage, options || { modal: false });
+        return this.addLogMessage(fullMessage);
+    }
+
+    /**
+     * Appends the message with a timestamp to the log channel
+     * @param message message to be added to the log channel
```

Comment:
The previous function docs have a blank line before their `@param` tags.

Applies to the following ones below and down.